lib/completion.js

```diff
@@ -19,8 +19,19 @@ module.exports = function (yargs, usage) {
     // to completion().
     if (completionFunction) {
       if (completionFunction.length < 3) {
+        var result = completionFunction(current, argv)
+
+        // promise based completion function.
+        if (typeof result.then === 'function') {
+          return result.then(function (list) {
+            process.nextTick(function () { done(list) })
```

Comment:
I'm perfectly fine with this, but (just out of curiosity) is the use of `process.nextTick()` necessary here? Does it help guard against `done(list)` throwing an error?